loopy/schedule/__init__.py

```diff
@@ -1649,6 +1649,10 @@ def filter_var_set_for_base_storage(var_name_set, base_storage_name):
 
                 race_var = race_var_base
 
+                if not src_race_vars:
```

Comment:
Unsure if this needs "tgt_race_vars == src_race_vars" criterion. Does anyone have a better intuition?

Reply:
I think it's OK as it is, in fact, I've just made the condition  even looser, so as to also bail out if no race variables are found on the target.